Modules/Order/Http/Services/OrderService.php

```diff
@@ -761,4 +876,168 @@ public function updateBilling(array $request): array
             'data' => ''
         ];
     }
-}
+
+    /**
+     * Create new order review
+     *
+     * @param array $reviewData
+     * @return array
+     */
+    public function storeReview(array $reviewData): array
+    {
+        $userId = auth()->user()->id;
+        $order = Order::where('order_number', $reviewData['order_number'])->first();
+
+        $orderReview = OrderReview::where('order_id', $order->id)->first();
+        // return error if already reviewed
+        if ($orderReview) {
+            return [
+                'res' => false,
+                'msg' => 'You have already reviewed !',
+                'data' => ""
+            ];
+        }
+        $reviewData['order_id'] = $order->id;
+        $reviewData['status'] = 'reviewed';
+        $reviewData['user_id'] = $userId;
+
+        //create review
+        $review = new OrderReview();
+        $review->fill($reviewData);
+        $review->save();
+
+        return [
+            'res' => true,
+            'msg' => 'Review posted Successfully',
+            'data' => $review
+        ];
+    }
+
+    /**
+     * Fetch list of return's policies.
+     *
+     * @return array
+     */
+    public function fetchReturnPolicies(): array
+    {
+        $returnPolicies = ReturnPolicy::where('status', 1)->get();
+
+        $response = ['res' => true, 'msg' => '', 'data' => $returnPolicies];
+
+        return ($response);
+    }
+
+    /**
+     * Fetch list of return's reasons.
+     *
+     * @return array
+     */
+    public function fetchReturnReasons(): array
+    {
+        $returnReasons = ReturnReason::where('status', 1)->get();
+
+        $response = ['res' => true, 'msg' => '', 'data' => $returnReasons];
+
+        return ($response);
+    }
+
+    /**
+     * Create new order return
+     *
+     * @param array $returnData
+     * @return array
+     */
+    public function createReturnOrder(array $returnData): array
+    {
+        $userId = auth()->user()->id;
+        $order = Order::where('order_number', $returnData['order_number'])->first();
+
+        $returnData['order_id'] = $order->id;
+        $policiesStr = '';
+        if (!empty($returnData['policy_values'])) {
+            $policiesStr = implode(',', $returnData["policy_values"]);
+        }
+        $returnData['policies'] = $policiesStr;
+        $returnData['shipping_date'] = date('Y-m-d', strtotime($returnData['shipping_time']));
```

Comment:
@arijit1111 Why we are sending shipping date for order return?

Reply:
it is returned pickup date